packages/mantine/src/components/collection/Collection.tsx

```diff
@@ -243,6 +243,7 @@ export const Collection = <T,>(props: CollectionProps<T> & {ref?: ForwardedRef<H
             draggable={draggable}
             onRemove={() => onRemoveItem?.(index)}
             removable={!(required && hasOnlyOneItem)}
+            testId={`item-${item.id}`}
```

Comment:
Since we're already sending the `id` on line 241 I would suggest to remove this prop and simply construct the `data-testid` in the `CollectionItem`

```tsx
data-testid={`item-${id}`}
```

My reasoning is the `Collection` doesn't need to care about the implementation of the `CollectionItem`